BlockAppealEmail.php

```diff
@@ -10,7 +10,7 @@ public static function onBlockAppealEmail(?string $body, array $args, Parser $pa
 		//disable caching so that the block information is truly up to date
 		$parser->getOutput()->updateCacheExpiry(0);
 		
-		if ($parser->getUser()->getBlock()) {
+		if ($parser->getUserIdentity()->getBlock()) {

```

Comment:
I don't think this code is valid?

Reply:
[Parser::getUserIdentity](https://doc.wikimedia.org/mediawiki-core/master/php/classParser.html#a9708965bf23d89bcbe031eaa3eb885d0) is a valid function according to the docs. It was added in MW 1.39 (which is why I'm PR'ing this into REL1_39 instead of master).